pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -663,6 +688,75 @@ func GetShortNameMode(ctx *types.SystemContext) (types.ShortNameMode, error) {
 	return config.shortNameMode, err
 }
 
+// GlobalCredentialHelpers returns the global top-level credential helpers.
+// A slice with only AuthenticationFileHelper is returned if no helpers are
+// configured.
+func GlobalCredentialHelpers(sys *types.SystemContext) ([]string, error) {
+	config, err := getConfig(sys)
+	if err != nil {
+		return nil, err
+	}
+	helpers := config.partialV2.CredentialHelpers
+	if len(helpers) == 0 {
+		helpers = append(helpers, AuthenticationFileHelper)
+	}
+	return helpers, nil
+}
+
+// FindCredentialHelpers returns a list of credential helpers for the
+// specified registry.  The helpers are intended to be contacted in the
+// returned order.
+//
+// Note that the global top-level credential helpers are ignored.
+func FindCredentialHelpers(sys *types.SystemContext, registry string) ([]string, error) {
+	reg, err := FindRegistry(sys, registry)
+	if err != nil {
+		return nil, err
+	}
+	if reg != nil && reg.CredentialHelpers != nil {
+		return reg.CredentialHelpers, nil
+	}
+	return []string{}, nil
+}
+
+// AllCredentialHelpers returns a list of all credential helpers listed either
+// as the top-level credential helpers (see `GlobalCredentialHelpers()` or
+// specific to registry (see `FindCredentialHelpers`).  The top-level
+// credential helpers are listed prior to the registry-specific ones, in the
+// specified order.
```

Comment:
Why does the order matter? If the caller doesn’t know the boundary, it isn’t a separation mechanism. Is it intended for some kind of prioritization? What does the order _mean_ for the caller, then? That earlier entries override the latter ones on conflict? The opposite?